test/e2e/deployment/rbac.go

```diff
@@ -0,0 +1,147 @@
+				clusterrolebinding := "clusterrolebinding" + "-" + UUIDgen.String()
+				podName := "pod" + "-" + UUIDgen.String()
+
+				//create serviceAccount
+				err = k8s.CreateServiceAccount(ctx, kubeClient, ns, serviceAccount)
+				Expect(err).To(BeNil())
+
+				//create pod with the serviceAccount
+				pod, err := k8s.CreatePodWithServiceAccount(kubeClient, ns, podName, serviceAccount)
+				log.Infof("pod with sa is %v\n", pod)
+				Expect(err).To(BeNil())
+
+				err = k8s.CreateRBACWithBindingSA(ctx, kubeClient, ns, serviceAccount, clusterRolename, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				_, err = k8s.GetClusterRoleBinding(ctx, kubeClient, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				//create backup for the rbac
+				backupName := "backup" + "rbac" + "-" + UUIDgen.String()
+				includeNs := "default"
+				resourceType := "Pod"
+				backup := kahu.NewBackup(backupName, includeNs, resourceType)
+				opts := metav1.CreateOptions{}
+
+				_, err = kahuClient.KahuV1beta1().Backups().Create(ctx, backup, opts)
+				Expect(err).To(BeNil())
+				log.Infof("backup of rbac is done\n")
+				time.Sleep(40 * time.Second)
+
+				// create restore for the backup
+				restoreName := "restore" + "rbac" + "-" + UUIDgen.String()
+				nsRestore := "restore-nsdemo"
+				restore := kahu.NewRestore(restoreName, nsRestore, includeNs, backupName)
+				log.Infof("restore is %v\n", restore)
+				restore1, err := kahuClient.KahuV1beta1().Restores().Create(ctx, restore, opts)
+				log.Debugf("restore1 is %v\n", restore1)
+				Expect(err).To(BeNil())
+				log.Infof("restore is created\n")
+
+				time.Sleep(40 * time.Second)
+
+				//check if the restored rbac is up
+				pod, err = k8s.GetPod(ctx, kubeClient, nsRestore, podName)
+				log.Infof("pod is %v\n", pod)
+				Expect(err).To(BeNil())
+
+				_, err = k8s.GetClusterRoleBinding(ctx, kubeClient, clusterrolebinding)
+				Expect(err).To(BeNil())
+				log.Infof("pod restored is up\n")
+
+				//Delete the. restore
+				optsDel := metav1.DeleteOptions{}
+				err = kahuClient.KahuV1beta1().Restores().Delete(ctx, restoreName, optsDel)
+				Expect(err).To(BeNil())
+
+				//Delete the backup
+				err = kahuClient.KahuV1beta1().Backups().Delete(ctx, backupName, optsDel)
+				Expect(err).To(BeNil())
+
+				//Delete SA created
+				err = k8s.DeleteServiceAccount(ctx, kubeClient, serviceAccount, ns)
+				Expect(err).To(BeNil())
+
+				err = k8s.DeleteServiceAccount(ctx, kubeClient, serviceAccount, nsRestore)
+				Expect(err).To(BeNil())
+
+				//Delete rbac created
+				err = k8s.CleanupClusterRoleBinding(ctx, kubeClient, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				//Delete restored rbac
+				err = k8s.CleanupClusterRoleBinding(ctx, kubeClient, clusterrolebinding)
+				Expect(err).To(BeNil())
+
+				//Delete pod
+				err = k8s.DeletePod(ctx, kubeClient, ns, podName)
+				Expect(err).To(BeNil())
+
+				err = k8s.DeletePod(ctx, kubeClient, nsRestore, podName)
+				Expect(err).To(BeNil())
```

Comment:
IMO, If all operation are done in a separate namespace. Deleting namespace will handle deleting all other resources

Reply:
done